File: lab-1/domain/album_manager.py
from domain.content_manager import ContentManager


class AlbumManager(ContentManager):
    def __init__(self, name):
        super().__init__(name)
        self.albums = []

    def add_item(self, album):
        self.albums.append(album)

    def remove_item(self, album):
        if album in self.albums:
            self.albums.remove(album)

    def display_contents(self):
        for album in self.albums:
            album.display_info()

File: src/main/kotlin/content_manager/implementations/AlbumManager.kt
package content_manager.implementations
import content_manager.ContentManager
import media_item.MediaItem
import media_item.implementations.Album

class AlbumManager(override val name: String) : ContentManager {
    val albums = mutableListOf<Album>()

    override fun addItem(item: MediaItem) {
        if (item is Album) {
            albums.add(item)
        }
    }

    override fun removeItem(item: MediaItem) {
        albums.remove(item)
    }

    override fun displayContents() {
        albums.forEach { it.displayInfo() }
    }
}

